### ISSUE TITLE
Cannot deserialize Dictionary<int, List<double>>

### ISSUE THREAD

**Issue by xmedeko** [mgholam/fastJSON#99]
fastJSON fails deserialize a field `Dictionary<int, List<double>>` or a field `Dictionary<int, double[]>` . I have `UseExtensions = false`. It fails with:
```
System.InvalidCastException: Object must implement IConvertible.
  at System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) [0x00055] in <1f0c1ef1ad524c38bbc5536809c46b48>:0 
  at fastJSON.deserializer.ChangeType (System.Object value, System.Type conversionType) [0x00246] in ...\fastJSON\JSON.cs:541 
  at fastJSON.deserializer.CreateDictionary (System.Collections.Generic.List`1[T] reader, System.Type pt, System.Type[] types, System.Collections.Generic.Dictionary`2[TKey,TValue] globalTypes) [0x000d9] in ...\fastJSON\JSON.cs:902 
  at fastJSON.deserializer.ParseDictionary (System.Collections.Generic.Dictionary`2[TKey,TValue] d, System.Collections.Generic.Dictionary`2[TKey,TValue] globaltypes, System.Type type, System.Object input) [0x00463] in ...\fastJSON\JSON.cs:738 
...
```

Note: serialization is ok.

**Comment by mgholam** [comment]
Thanks, I will look into it.

**Comment by mgholam** [comment]
The following works for me, what is your test case?
```c#
        var d = new Dictionary<string, List<double>>();
        d.Add("a", new List<double> { 1, 2, 3 });
        d.Add("b", new List<double> { 4, 5, 6 });
        var s = JSON.ToJSON(d, new JSONParameters { UseExtensions = false });

        var o = JSON.ToObject<Dictionary<string, List<double>>>(s);
```

**Comment by xmedeko** [comment]
Sorry, it's `Dictionary<int, List<double>>` and Dictionary<int, double[]>` what is failing. (I have originally enum as a key in my code.)

**Comment by mgholam** [comment]
fixed in v2.2.3
